packages/flutter/test/widgets/slivers_appbar_floating_pinned_test.dart

```diff
@@ -6,7 +6,203 @@ import 'package:flutter/rendering.dart';
 import 'package:flutter_test/flutter_test.dart';
 import 'package:flutter/material.dart';
 
+import 'semantics_tester.dart';
+
 void main() {
+  testWidgets('Sliver appbars - floating and pinned - correct semantics', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      Localizations(
+        locale: const Locale('en', 'us'),
+        delegates: const <LocalizationsDelegate<dynamic>>[
+          DefaultWidgetsLocalizations.delegate,
+          DefaultMaterialLocalizations.delegate,
+        ],
+        child: Directionality(
+          textDirection: TextDirection.ltr,
+          child: MediaQuery(
+            data: const MediaQueryData(),
+            child: CustomScrollView(
+              slivers: <Widget>[
+                const SliverAppBar(
+                  title: Text('Hello'),
+                  pinned: true,
+                  floating: true,
+                  expandedHeight: 200.0,
+                ),
+                SliverFixedExtentList(
+                  itemExtent: 100.0,
+                  delegate: SliverChildBuilderDelegate(
+                    (BuildContext c, int index) {
```

Comment:
nit: the convention is to name unused parameters with `_` instead of just a single letter.

```suggestion
                    (BuildContext _, int index) {
```